cmake/modules/AddSwift.cmake

```diff
@@ -131,7 +131,8 @@ function(_add_host_variant_c_compile_flags target)
 
   is_build_type_optimized("${CMAKE_BUILD_TYPE}" optimized)
   if(optimized)
-    target_compile_options(${target} PRIVATE -O2)
+    clang_optimize_flag_for_build_type("${CMAKE_BUILD_TYPE}" optimize_flag)
+    target_compile_options(${target} PRIVATE "${optimize_flag}")
```

Comment:
Why is this needed?  Why is the default not sufficient?  Why can `${CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE}}` not sufficient?

Reply:
So I think you're pointing out the right thing, but there's at least one behavior change if we did that: `CMAKE_CXX_FLAGS_RELEASE` is -O3, whereas we today do -O2. How about I send out a separate PR with to try to improve this bit here?

Reply:
(I don't want to change any observable behavior of any other preset with this PR.)